File: src/main/java/alz/genfive/webshop/entity/Product.java
package alz.genfive.webshop.entity;

import alz.genfive.webshop.enumeration.Availability;
import alz.genfive.webshop.enumeration.Category;
import jakarta.persistence.*;
import lombok.*;

import java.util.List;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@ToString
@Entity
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(nullable = false, updatable = false)
    private Long id;
    private String name;
    private String description;
    private Float price;
    private Category category;
    @OneToMany
    private List<Review> reviews;
    private Availability availability;
    private String image;

}

File: webshop_frontend_angular/src/app/entity/Product.ts
import { Availability } from "../enumeration/Availability";
import { Category } from "../enumeration/Category";
import { List } from "../utils/List";
import { Review } from "./Review";

export interface Product{
    id: number;
    name: string;
    description: string;
    price: number;
    salePrice: number;
    sale: boolean;
    category: Category;
    reviews: List<Review>;
    availability: Availability;
    imageSource: string;
}